drivers/media/platform/bcm2835/bcm2835-camera.c

```diff
@@ -1792,8 +1895,10 @@ static int __init bm2835_mmal_init(void)
 free_dev:
 	kfree(dev);
 
-	v4l2_err(&dev->v4l2_dev,
-		 "%s: error %d while loading driver\n",
+	for ( ; camera > 0; camera--)
+		bcm2835_cleanup_instance(gdev[camera]);
```

Comment:
Possibly. 
This is failure to load, so won't that unload the module due to the failure and therefore deallocate gdev.

Probably safest to reset it to NULL anyway - I'll update it.